File: leetcode_practice/src/2520_统计能整除数字的位数.rs
/*
 * @lc app=leetcode.cn id=2520 lang=rust
 *
 * [2520] 统计能整除数字的位数
 */

 /**
    Accepted 2023-01-03
    56/56 (0 ms)
    Your runtime beats 100 % of rust submissions
    Your memory usage beats 88.57 % of rust submissions (2 MB)
  */
// @lc code=start
impl Solution {
    pub fn count_digits(num: i32) -> i32 {
        let mut n = num;
        let mut res = 0;

        while 0 != n {
            if 0 == num % (n % 10 ) {
                res += 1;
            }
            n /= 10;
        }
        res
    }
}
// @lc code=end

File: cpp/2520_统计能整除数字的位数.cpp
/*
 * @lc app=leetcode.cn id=2520 lang=cpp
 *
 * [2520] 统计能整除数字的位数
 */

#include "AllInclude.h"

// @lc code=start
/**
 * Accepted 2023-10-26
 * 56/56 (0 ms)
 * Your runtime beats 100 % of cpp submissions
 * Your memory usage beats 26.67 % of cpp submissions (6.2 MB)
*/
class Solution {
public:
    int countDigits(int num) {
        int tmp = num, res = 0;
        while (tmp) {
            int r = tmp % 10;
            if (!(num % r)) {
                ++res;
            }
            tmp /= 10;
        }
        return res;
    }
};
// @lc code=end

